src/Analysis/Engine/Test/AstAnalysisTests.cs

```diff
@@ -381,91 +377,91 @@ class BankAccount(object):
                     .Which.Should().HaveNestedModule("Module")
-                    .Which.Should().HaveClass("Class")
-                    .Which;
+                    .Which.Should().HaveMultipleTypesMember("Class"); // member information comes from multiple sources
 
                 analysis.Should().HaveVariable("c")
                     .WithValue<IBuiltinInstanceInfo>()
-                    .Which.Should().HaveMemberType(PythonMemberType.Instance)
-                    .And.HavePythonType(type)
                     .Which.Should().HaveMembers("untyped_method", "inferred_method", "typed_method_2")
                     .And.NotHaveMembers("typed_method");
             }
         }
 
         [TestMethod, Priority(0)]
-        [Ignore("https://github.com/Microsoft/python-language-server/issues/59")]
         public async Task AstTypeStubPaths_ExclusiveStubs() {
             using (var server = await CreateServerAsync()) {
-                var uri = await server.OpenDefaultDocumentAndGetUriAsync("import Package.Module\n\nc = Package.Module.Class()");
-                await server.GetAnalysisAsync(uri);
-
-                server.Analyzer.Limits = new AnalysisLimits {
-                    UseTypeStubPackages = true,
-                    UseTypeStubPackagesExclusively = true
-                };
-                server.Analyzer.SetTypeStubPaths(new[] { TestData.GetPath("TestData\\AstAnalysis\\Stubs") });
-                server.EnqueueItem(uri);
-
-                var analysis = await server.GetAnalysisAsync(uri);
-
-                var type = analysis.Should().HavePythonModuleVariable("Package")
-                    .Which.Should().HaveNestedModule("Module")
-                    .Which.Should().HaveClass("Class")
-                    .Which;
+                var analysis = await GetStubBasedAnalysis(
+                    server, 
+                    "import Package.Module\n\nc = Package.Module.Class()",
+                    new AnalysisLimits {
+                        UseTypeStubPackages = true,
+                        UseTypeStubPackagesExclusively = true
+                    },
+                    searchPaths: new[] { TestData.GetPath("TestData\\AstAnalysis") },
+                    stubPaths: new[] { TestData.GetPath("TestData\\AstAnalysis\\Stubs") });
+
+                analysis.Should().HavePythonModuleVariable("Package");
 
                 analysis.Should().HaveVariable("c")
                     .WithValue<IBuiltinInstanceInfo>()
-                    .Which.Should().HaveMemberType(PythonMemberType.Instance)
-                    .And.HavePythonType(type)
-                    .Which.Should().HaveMethod("typed_method_2")
+                    .Which.Should().HaveMembers("typed_method_2")
                     .And.NotHaveMembers("untyped_method", "inferred_method", "typed_method");
             }
         }
 
+        private async Task<IModuleAnalysis> GetStubBasedAnalysis(
```

Comment:
It doesn't look like this method has reduced the amount of code.

Reply:
It is more if we want to add additional tests with actual multiple paths and stubs. Existing one are rather primitive.

Reply:
Is it actually possible in VSC to change these parameters (Limits, SearchPaths) after server is started?

Reply:
Currently no. LS restart is required.

Reply:
In this case, maybe we should simplify these tests by setting all these values from the beginning? We can add another overload of `ServerExtensions.InitializeAsync`.

Reply:
But is actually good to set them later. LS doesn't do it for simplicity but it really should not be requiring restart. So at some point we should handle change.

Reply:
Ok